src/commands/giveaway.ts

```diff
@@ -1,54 +1,184 @@
+        const numWinners = interaction.options.getInteger("winners") ?? 1;
+
+        const guild = interaction.guild;
+
+        const channel = guild.channels.cache.get(this.announcementChannelId);
+
+        const giveawayMessage = new EmbedBuilder()
+            .setColor(0x0099FF)
+            .setTitle(`**${prize}**`)
+            .setDescription(`Number of Winners: ${numWinners}\n Prize: ${prize}`)
+		
+
+        const enterGiveawayButton = new ButtonBuilder()
+            .setCustomId("enterGiveaway")
+            .setLabel("Enter Giveaway")
+            .setStyle(ButtonStyle.Primary);
+
+        const row = new ActionRowBuilder()
+            .addComponents(enterGiveawayButton);
+
+        await interaction.reply(
+            {
+                content: `Giveaway successfully created for ${role} with ${numWinners} winners for ${duration} seconds, in ${channel}`,
+                ephemeral: true
+            }
+        );
+
+        const response = await channel.send(
+            {
+                content: `🎉 Giveaway Started! 🎉`,
+                embeds: [giveawayMessage],
+                components: [row],
+            }
+        );
+
+        const collector = response.createMessageComponentCollector({
+            componentType: ComponentType.Button,
+            time: duration,
+        });
+
+        collector.on("collect", async (cInteraction: ButtonInteraction) => {
+            if (cInteraction.customId === "enterGiveaway") {
+
+                const giveawayUser = cInteraction.user;
+                const userId = giveawayUser.id;
+
+                if (!participants.includes(userId)) {
+
+                    participants.push(userId);
+                    await cInteraction.reply({
+                        content: "You have entered the giveaway.",
+                        ephemeral: true
+                    });
+                } else {
+                    await cInteraction.reply({
+                        content: "You have already entered the giveaway.",
+                        ephemeral: true
+                    });
+                }
+            }
+        });
+
+        collector.on("end", async () => {
+
+            enterGiveawayButton.setDisabled(true);
+
+            if (participants.length === 0) {
+
+                const noResponse = new EmbedBuilder()
+                    .setColor(0xFF0000)
+                    .setTitle(`**${prize}**`)
+                    .setDescription("No one participated in the giveaway. :frowning:");
+
+                await response.edit({
+                    components: [row]
+                });
+            
+                await channel.send({
+                    embeds: [noResponse]
+                });
+                return;
+            }
+
+            const winners = [];
+            for (let i = 0; i < numWinners; i++) {
+                const winnerIndex = Math.floor(Math.random() * participants.length);
```

Comment:
This is not random

Reply:
Maalni, we won't be using this PR anymore. We merged light's PR because my branch wasn't working.

If you want me to change it, please DM and check the file via the main branch.